### PR TITLE
RFR - CW Special Pigeon Sandwich Maker Bot

### PR DESCRIPTION
**Pull request by dthyresson**
Port of the CW Specials Pigeon http://codepen.io/dthyresson/full/xGEyzj/ to KevinBot.

Some niceties include:

* menu of sides, mains, breads are now in `YAML` as opposed to arrays of strings
* ActiveSupport provides `.to_sentence`

This PR includes some KevinBot changes:

* If logging is on, then outputs errors and response to standard out
* RSpec so can now write specs for your custom commands
* Can define option custom username and icon_url for slack response
*  `classify` the command

### CODE REVIEW

File: commands/pigeon.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  def main
+    @main ||= menu[:mains].sample
+  end
+
+  def sample_sides
+    @sample_sides ||= menu[:sides].sample(rand(1..4))
+  end
+
+  def sides(number_of_sides = 4)
```
Comment (edwardloveall) [inline commands/pigeon.rb]: Do you use this variable anywhere?

File: commands/pigeon.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      return sample_sides.to_sentence
+    end
+  end
+
+  def bread_jective
+    @bread_jective ||= menu[:bread_jectives].sample
+  end
+
+  def bread
+    @bread ||=  if bread_jective.blank?
```
Comment (edwardloveall) [inline commands/pigeon.rb]: Extra space between the `=` and `if`

File: kevinbot.rb [hunk 3]
```diff
@@ -9,19 +9,34 @@
 Dir['commands/**/*.rb'].each { |f| load f }
 
 post '/' do
-  if ENV['LOG'] == 'true'
-    puts params
-  end
+  puts params if log?
```
Comment (edwardloveall) [inline kevinbot.rb]: Smart. This was always a weird bit of code

File: commands/pigeon.rb (hunk 1 above)
Comment (dthyresson) [inline commands/pigeon.rb]: Ah, yes, it is meant to be a param to

`@sample_sides ||= menu[:sides].sample(rand(1..4))` <- the 4

Comment (edwardloveall) [inline commands/pigeon.rb]: DT,

Do I need to do anything with these?

File: commands/pigeon.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    if (sample_sides.length == 1)
+      "#{menu[:side_jectives].sample} #{sample_sides.first}"
+    else
+      sample_sides.to_sentence
+    end
+  end
+
+  def bread_jective
+    @bread_jective ||= menu[:bread_jectives].sample
+  end
```
Comment (hchood) [inline commands/pigeon.rb]: This may be my favorite method name ever.

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+Dir[File.join(File.dirname(__FILE__), '..', '**.rb')].each do |f|  require f
+end
```
Comment (hchood) [inline spec/spec_helper.rb]: use curlies to oneline the block?

**Pull request merged**